app/controllers/api/profile_controller.rb

```diff
@@ -5,6 +5,14 @@ def show
     respond_with_resource serializer: UserSerializer
   end
 
+  def mentionable_users
+    @users = resource_class.mentionable_by(current_user).search_for(params[:q])
```

Comment:
I'd argue that you want to run this through `instantiate_collection` so that it pages; no point in returning more than 30 results here; in fact we could probably get away with first 5/10.

```
instantiate_collection do
  collection.mentionable_by(current_user).search_for(params[:q])
end
respond_with_collection serializer: Simple::UserSerializer
```

what trouble are you running into with respond_with_collection?

Reply:
Should be something like

```
def mentionable_users
  instantiate_collection { |collection| collection.search_for(params[:q]) }
  respond_with_collection serializer: Simple::UserSerializer, root: :users
end

def accessible_records
  resource_class.mentionable_by(current_user)
end
```

Reply:
When I use respond_with_collection it always returns current_user, I couldn't see how that was being done.

Reply:
Oh right, it's because 'resource' is always defined in the profile controller (simplified version:)

```
def resource
  current_user 
end

def resources_to_serialize
  Array (resource || collection)
end
```

I'm not certain why I picked resource first; it's probably arbitrary, but for now I'd override `resources_to_serialize` to be `collection || resource`, and take a look later at seeing if we could update it in restful_controller

Reply:
I initially did see that, try it, but it did not work.... and now I don't know why, because now it does. Anyway, thanks!